controllers/managedocs_controller.go

```diff
@@ -1043,14 +1045,22 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 			}
 		}
 
-		r.prometheus.Spec = desired.Spec
-		r.prometheus.Spec.Alerting.Alertmanagers[0].Namespace = r.namespace
-		r.prometheus.Spec.AdditionalAlertRelabelConfigs = &corev1.SecretKeySelector{
-			LocalObjectReference: corev1.LocalObjectReference{
-				Name: alertRelabelConfigSecretName,
-			},
-			Key: alertRelabelConfigSecretKey,
-		}
+		// getting clusterID
+        clusterVersion := &configv1.ClusterVersion{}
+        clusterVersion.Name = "version"
+        if err := r.unrestrictedGet(clusterVersion); err != nil {
+            return err
+        }
+
+        r.prometheus.Spec = desired.Spec
+        r.prometheus.Spec.ExternalLabels["clusterId"] = string(clusterVersion.Spec.ClusterID)
+        r.prometheus.Spec.Alerting.Alertmanagers[0].Namespace = r.namespace
+        r.prometheus.Spec.AdditionalAlertRelabelConfigs = &corev1.SecretKeySelector{
+            LocalObjectReference: corev1.LocalObjectReference{
+                Name: alertRelabelConfigSecretName,
+            },
+            Key: alertRelabelConfigSecretKey,
+        }
```

Comment:
Something is wrong with the indentation here

Reply:
corrected